libavfilter/vulkan.c

```diff
@@ -472,33 +472,36 @@ int ff_vk_add_exec_dep(AVFilterContext *avctx, FFVkExecContext *e,
     int planes = av_pix_fmt_count_planes(fc->sw_format);
 
     for (int i = 0; i < planes; i++) {
-        e->sem_wait = av_fast_realloc(e->sem_wait, &e->sem_wait_alloc,
-                                      (e->sem_wait_cnt + 1)*sizeof(*e->sem_wait));
-        if (!e->sem_wait) {
-            ff_vk_discard_exec_deps(avctx, e);
-            return AVERROR(ENOMEM);
-        }
+        if (input_frame) {
+            e->sem_wait = av_fast_realloc(e->sem_wait, &e->sem_wait_alloc,
+                                        (e->sem_wait_cnt + 1)*sizeof(*e->sem_wait));
+            if (!e->sem_wait) {
+                ff_vk_discard_exec_deps(avctx, e);
+                return AVERROR(ENOMEM);
+            }
 
-        e->sem_wait_dst = av_fast_realloc(e->sem_wait_dst, &e->sem_wait_dst_alloc,
-                                          (e->sem_wait_cnt + 1)*sizeof(*e->sem_wait_dst));
-        if (!e->sem_wait_dst) {
-            ff_vk_discard_exec_deps(avctx, e);
-            return AVERROR(ENOMEM);
-        }
+            e->sem_wait_dst = av_fast_realloc(e->sem_wait_dst, &e->sem_wait_dst_alloc,
+                                            (e->sem_wait_cnt + 1)*sizeof(*e->sem_wait_dst));
+            if (!e->sem_wait_dst) {
+                ff_vk_discard_exec_deps(avctx, e);
+                return AVERROR(ENOMEM);
+            }
 
-        e->sem_sig = av_fast_realloc(e->sem_sig, &e->sem_sig_alloc,
-                                     (e->sem_sig_cnt + 1)*sizeof(*e->sem_sig));
-        if (!e->sem_sig) {
-            ff_vk_discard_exec_deps(avctx, e);
-            return AVERROR(ENOMEM);
-        }
+            e->sem_wait[e->sem_wait_cnt] = f->sem[i];
+            e->sem_wait_dst[e->sem_wait_cnt] = in_wait_dst_flag;
+            e->sem_wait_cnt++;
+        } else {
 
-        e->sem_wait[e->sem_wait_cnt] = f->sem[i];
-        e->sem_wait_dst[e->sem_wait_cnt] = in_wait_dst_flag;
-        e->sem_wait_cnt++;
+            e->sem_sig = av_fast_realloc(e->sem_sig, &e->sem_sig_alloc,
```

Comment:
Maybe you can send other patch to the realloc issue. 
the ralloc usage is wrong?
when realloc returns NULL, it will not free original memory.  This code will leak memory at that time

Reply:
I think the semaphore usage is wrong. You can see the previous code. The semaphores are set to both wait and signal. I don't change its' way to use realloc, I rearrange them to set watSem and signalSem separately. 

The original resource will be free in filter uninit() function. I think program will go into uninit() function if function returning error.

Reply:
uninit can't free the memory since you already set e->sem_sig to NULL.

Reply:
You are right. My bad.
Now I have fixed it.
Thanks a lot.